pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/CountAggregationFunction.java

```diff
@@ -30,34 +31,48 @@
 import org.apache.pinot.core.query.aggregation.groupby.GroupByResultHolder;
 import org.apache.pinot.segment.spi.AggregationFunctionType;
 import org.apache.pinot.segment.spi.index.startree.AggregationFunctionColumnPair;
+import org.roaringbitmap.RoaringBitmap;
 
 
-public class CountAggregationFunction implements AggregationFunction<Long, Long> {
-  private static final String COLUMN_NAME = "count_star";
-  private static final String RESULT_COLUMN_NAME = "count(*)";
+public class CountAggregationFunction extends BaseSingleInputAggregationFunction<Long, Long> {
   private static final double DEFAULT_INITIAL_VALUE = 0.0;
   // Special expression used by star-tree to pass in BlockValSet
   private static final ExpressionContext STAR_TREE_COUNT_STAR_EXPRESSION =
       ExpressionContext.forIdentifier(AggregationFunctionColumnPair.STAR);
 
+  private final String _columnName;
+  private final String _resultColumnName;
+  private final boolean _nullHandlingEnabled;
+
+  public CountAggregationFunction(ExpressionContext expression, boolean nullHandlingEnabled) {
+    super(expression);
+    String expStr = expression.toString();
```

Comment:
When null handling is enabled, we can set the argument to `*` to keep the current behavior

Reply:
Null handling requires a single column expression. We cannot handle nulls in the star expression case.

Reply:
Sorry my bad. What I meant is when null handling is **disabled**, we can set it to `*`